File: Semestrul 4/Medii de proiectare si programare/Laboratoare/FirmaTransportC#/Services/ITransportService.cs
using FirmaTransportC_.Model;

namespace FirmaTransportC_.Services
{
    public interface ITransportService
    {
        void Login(User user, IObserver client);
        void Logout(User user);
        List <Trip> GetAllTrips();
        List<Trip> SearchTrips(Trip trip);
        List<Seat> GetSeatsForTrip(Trip trip);
        void MakeReservation(Reservation reservation);
        void CancelReservation(Reservation reservation);
    }
}

File: Semestrul 4/Medii de proiectare si programare/Laboratoare/FirmaTransportJava/Services/src/main/java/transport/services/ITransportService.java
package transport.services;

import transport.model.Reservation;
import transport.model.Seat;
import transport.model.Trip;
import transport.model.User;

import java.util.List;

public interface ITransportService {

    void login(User user, IObserver client) throws ServiceException;

    void logout(User user) throws ServiceException;

    List<Trip> getAllTrips() throws ServiceException;

    List<Trip> searchTrips(Trip trip) throws ServiceException;

    List<Seat> getSeatsForTrip(Trip trip) throws ServiceException;

    void makeReservation(Reservation reservation) throws ServiceException;

    void cancelReservation(Reservation reservation) throws ServiceException;
}
